app/controllers/api/v1/bulk_export_controller.rb

```diff
@@ -17,60 +17,55 @@ def index
-            bulk_export.tei_work = !(params[:tei_work].blank?)
-            bulk_export.html_page = !(params[:html_page].blank?)
-            bulk_export.html_work = !(params[:html_work].blank?)
-            bulk_export.subject_csv_collection = !(params[:subject_csv_collection].blank?)
-            bulk_export.subject_details_csv_collection = !(params[:subject_details_csv_collection].blank?)
-            bulk_export.table_csv_collection = !(params[:table_csv_collection].blank?)
-            bulk_export.table_csv_work = !(params[:table_csv_work].blank?)
-            bulk_export.notes_csv = !(params[:notes_csv].blank?)
-            bulk_export.save
-            bulk_export.submit_export_process
+        if collection.show_to?(@api_user)
+          # try to parse the bulk_export
+          bulk_export = BulkExport.new
+          bulk_export.collection = collection
+          bulk_export.user = @api_user
+          bulk_export.status = BulkExport::Status::NEW
+          bulk_export.plaintext_verbatim_page = !params[:plaintext_verbatim_page].blank?
+          bulk_export.plaintext_verbatim_work = !params[:plaintext_verbatim_work].blank?
+          bulk_export.plaintext_emended_page = !params[:plaintext_emended_page].blank?
+          bulk_export.plaintext_emended_work = !params[:plaintext_emended_work].blank?
+          bulk_export.plaintext_searchable_page = !params[:plaintext_searchable_page].blank?
+          bulk_export.plaintext_searchable_work = !params[:plaintext_searchable_work].blank?
+          bulk_export.tei_work = !params[:tei_work].blank?
+          bulk_export.html_page = !params[:html_page].blank?
+          bulk_export.html_work = !params[:html_work].blank?
+          bulk_export.subject_csv_collection = !params[:subject_csv_collection].blank?
+          bulk_export.subject_details_csv_collection = !params[:subject_details_csv_collection].blank?
+          bulk_export.table_csv_collection = !params[:table_csv_collection].blank?
+          bulk_export.table_csv_work = !params[:table_csv_work].blank?
+          bulk_export.notes_csv = !params[:notes_csv].blank?
+          bulk_export.save
+          bulk_export.submit_export_process
 
-            response = {
-              :id => bulk_export.id,
-              :status => bulk_export.status,
-              :status_uri => api_v1_bulk_export_status_url(bulk_export.id),
-              :download_uri => api_v1_bulk_export_download_url(bulk_export.id)
-            }
+          response = {
+            id: bulk_export.id,
+            status: bulk_export.status,
+            status_uri: api_v1_bulk_export_status_url(bulk_export.id),
+            download_uri: api_v1_bulk_export_download_url(bulk_export.id)
+          }
 
-            render status: 202, json: response.to_json
+          render status: 202, json: response.to_json
 
-          else
-            render status: 403, json: 'User #{@api_user} is not authorized to view #{collection.title}'
```

Comment:
This is one of the status typo. We are not interpolating the string here due to single quotes

Reply:
Let's fix it!